src/configuration/services/network-storage.md

```diff
@@ -0,0 +1,187 @@
+
+## Supported versions
+
+* 1.0
+
+(This is a reference to a version of our network storage implementation, not to a version of a 3rd party application.)
+
+## Define the service
+
+First, declare a new service in the `services.yaml` file like so:
+
+```yaml
+files:
+    type: network-storage:1.0
+    disk: 2048
+```
+
+This example creates a service named `files` that is of type `network-storage`, and gives it 2048 MB of storage total.
+
+## Declare the mount
+
+Second, add the following entry to your mounts list:
+
+```yaml
+mounts:
+    'my/files':
+        source: service
+        service: files
+        source_path: network
+```
+
+This block will declare a writeable mount on the application container at the path `my/files`, which will be provided by the `files` service defined above.  The `source_path` specifies the path within the network service that the mount points to.  It is often easiest to have it match the name of the mount point itself but that is not required.
+
+Note that you do *not* need to add a relationship to point to the `files` service.  That is handled automatically by the system.
+
+The application container can now read from and write to the `my/files` path just as if it were a local writeable mount.
+
+## Multi-application usage
+
+If your project contains more than one application (that is, multiple directories with their own `.platform.app.yaml` files), they can all use the same network mounts if desired.  If the `source_path` is the same in both `.platform.app.yaml` files then the files will be shared between both applications, even if the mount location is different.
+
+It is also possible to have one application mount a `source_path` that is a subdirectory of another application's mount.  For example:
+
+app1:
+
+```yaml
+mounts:
+    'web/uploads':
+        source: service
+        service: files
+        source_path: uploads
+```
+
+app2:
+
+```yaml
+mounts:
+    'process':
+        source: service
+        service: files
+        source_path: uploads/incoming
+    'done':
+        source: service
+        service: files
+        source_path: uploads/done
+```
+
+In this example, `app1` will have access to the entire `uploads` directory by writing to `web/uploads`.  `app2`, by contrast, will have two mounts that it can write to: `process` and `done`.  The `process` mount will refer to the same directory as the `web/uploads/incoming` directory does on `app1`, and the `done` mount will refer to the same directory as the `web/uploads/done` directory on `app1`.
+
+## Worker instances
+
+When defining a [Worker](/configuration/app/workers.md) instance it is important to keep in mind what mount behavior is desired.  Unless the `mounts` block is defined within the `web` and `workers` sections separately, a top level `mounts` block will apply to both instances.  However, `local` mounts will be separate for each instance while `service` mounts will refer to the same file system.  For example:
```

Comment:
> it is important to keep in mind what mount behavior is desired.

This reads weird, no? That feels like it should be "which mount behavior is desired"?

Reply:
Also:

> However, `local` mounts will be separate for each instance

That feels like it's not explaining correctly. Something like "However, `local` mounts will be separate disks for each instance", or something along these lines? I'm not sure I like my way either.

Reply:
"independent for each instance"?  Like, what's the opposite of shared? :smile:

Reply:
dunno, you're the native :P

Reply:
Looking at the context again, I think "what" is correct here.  "Which" implies an either/or type situation, whereas what we offer is a "mix and match" configuration.